llvm/include/llvm/Transforms/Instrumentation/CFGMST.h

```diff
@@ -291,10 +301,14 @@ template <class Edge, class BBInfo> class CFGMST {
     return *AllEdges.back();
   }
 
-  CFGMST(Function &Func, bool InstrumentFuncEntry,
+  CFGMST(Function &Func, bool InstrumentFuncEntry, bool InstrumentLoopEntries,
          BranchProbabilityInfo *BPI = nullptr,
-         BlockFrequencyInfo *BFI = nullptr)
-      : F(Func), BPI(BPI), BFI(BFI), InstrumentFuncEntry(InstrumentFuncEntry) {
+         BlockFrequencyInfo *BFI = nullptr, LoopInfo *LI = nullptr)
+      : F(Func), BPI(BPI), BFI(BFI), LI(LI),
+        InstrumentFuncEntry(InstrumentFuncEntry),
+        InstrumentLoopEntries(InstrumentLoopEntries) {
+    assert(!(InstrumentLoopEntries && !LI) &&
+           "expected a LoopInfo to instrumenting loop entries");
     buildEdges();
     sortEdgesByWeight();
     computeMinimumSpanningTree();
```

Comment:
I wonder if it makes sense to add an assert here that each loop entry will be instrumented.

Reply:
Could you elaborate what exactly you mean by this? Assert in the end that all loop entries are instrumented or what exactly?

Reply:
If `InstrumentLoopEntries` is true, then each loop entry should be instrumented. I'm asking if we should add a compile time assert that indeed each loop entry will be instrumented after we compute the minimum spanning tree.